invenio_app_ils/search/permissions.py

```diff
@@ -0,0 +1,117 @@
+        return Q()
+
+    # A record is public if `open_access` field True or missing
+    open_access_field_missing = ~Q("exists", field="open_access")
+    is_open_access = open_access_field_missing | Q("term", open_access=True)
+
+    combined_filter = is_open_access
+
+    if current_app.config.get("ILS_RECORDS_EXPLICIT_PERMISSIONS_ENABLED"):
+        # if `_access`, check `_access.read` against the user. It takes
+        # precedence over `open_access`.
+        # if not `_access`, check if open access as before.
+        _access_field_exists = Q("exists", field="_access.read")
+        provides = _get_user_provides()
+        user_can_read = _access_field_exists & Q(
+            "terms", **{"_access.read": provides}
+        )
+        combined_filter = user_can_read | (
+            ~_access_field_exists & is_open_access
+        )
+
+    return Q("bool", filter=[combined_filter])
+
+
+def _ils_search_factory(self, search, qs_validator):
+    """Search factory with Query String validator.
+
+    :param self: REST view.
+    :param search: Elastic search DSL search instance.
+    :returns: Tuple with search instance and URL arguments.
+    """
+    def query_parser(qstr=None):
+        """Default parser that uses the Q() from elasticsearch_dsl."""
+        if qstr:
+            return Q('query_string', query=qstr)
+        return Q()
+
+    from invenio_records_rest.facets import default_facets_factory
+    from invenio_records_rest.sorter import default_sorter_factory
+
+    query_string = request.values.get("q")
+
+    if not current_user.is_authenticated:
+        raise UnauthorizedSearchError(query_string)
+
+    query = query_parser(qs_validator(query_string))
+
+    try:
+        search = search.query(query)
+    except SyntaxError:
+        raise SearchQueryError(query_string)
+
+    search_index = search._index[0]
+    search, urlkwargs = default_facets_factory(search, search_index)
+    search, sortkwargs = default_sorter_factory(search, search_index)
+    for key, value in sortkwargs.items():
+        urlkwargs.add(key, value)
+
+    urlkwargs.add("q", query_string)
+    return search, urlkwargs
+
+
+def search_factory_filter_by_patron(self, search):
+    """Prepare query string to filter records by current logged in user."""
+
+    def filter_by_patron(query_string):
+        """Filter search results by patron_pid."""
+        # if the logged in user in not librarian or admin, validate the query
+        if has_request_context() and not backoffice_permission().allows(
+            g.identity
+        ):
+            # patron can find only his records
+            if not query_string:
+                # force query to be patron_pid:<logged in user>
+                query_string = "patron_pid:{}".format(g.identity.id)
+            else:
+                # check for patron_pid query value
+                match = re.match(r"patron_pid:(?P<pid>\d)", query_string)
```

Comment:
```suggestion
                match = re.match(r"patron_pid:(?P<pid>\d+)", query_string)
```
Or else it would only match with the first digit of the patron ID.

Reply:
Now I am thinking, is patron_pid alphanumeric? @ntarocco

Reply:
it is not for now, it does not pass through invenio-pidstore, no?

Reply:
Well spotted! Actually I just moved the code from one place to another. I have applied the fix.
Yes user ids (int) at the moment.